contracts/staking/StakingPoolLib.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library StakingPoolLib {
+    enum StakingPoolType {
+        FIXED,
```

Comment:
question: is the default pool type `FIXED`?

As Solidity has a default value of zero for enum references, when no value is set to a pointer it'll hold zero as the value, which for this enum will be `FIXED`.
Using `UNINITIALIZED` as the first element in an enum is an approach to mitigate the risk of the enum not boing set.
Does it make sense to have `FIXED` as the pool type if the contract does not set the reference?